src/Fmt_ast.ml

```diff
@@ -607,23 +607,25 @@ let fmt_docstring c ?(standalone = false) ?pro ?epi doc =
         | _, false -> false
       in
       let try_parse str_cmt =
-        let if_fail = if c.conf.wrap_comments then fill_text else str in
         match Octavius.parse (Lexing.from_string str_cmt) with
-        | Error _ -> if_fail str_cmt
+        | Error _ ->
+            (if c.conf.wrap_comments then fill_text else str) str_cmt
         | Ok parsed ->
             if Conf.debug then (
               Octavius.print Caml.Format.str_formatter parsed ;
               Caml.Format.eprintf "%s%!"
                 (Caml.Format.flush_str_formatter ()) ) ;
-            let fmted = Fmt_odoc.fmt parsed in
-            let spaces = ['\t'; '\n'; '\011'; '\012'; '\r'; ' '] in
-            let is_space = List.mem ~equal:Char.equal spaces in
-            let space_0 = try is_space str_cmt.[0] with _ -> false in
-            let space_n =
-              try is_space str_cmt.[String.length str_cmt - 1] with _ ->
-                false
+            let space_i i =
+              let spaces = ['\t'; '\n'; '\011'; '\012'; '\r'; ' '] in
+              let f = List.mem ~equal:Char.equal spaces in
+              let default = false in
+              Option.value_map
+                (List.nth (String.to_list str_cmt) i)
+                ~default ~f
```

Comment:
Maybe I'm missing something, but why not save the allocation and list searching with:
```
0 <= i && i < String.length str_cmt && is_space str_cmt.[i]
```

Reply:
you are right, this is better